packages/loader/driver-definitions/src/urlResolver.ts

```diff
@@ -38,6 +38,26 @@ export interface IUrlResolver {
     ): Promise<string>;
 }
 
+/**
+* Information that can be returned by a lightweight, seperately exported driver function. Used to preanalyze a URL
+* for driver compatibility and preload information.
+*/
+export interface DriverPreCheckInfo {
+    /**
+     * Only needed as long as long as Loader.request() does not work as intended. When Loader.request() caches and
+     * resolves pathing properly, this can be removed. #4489, #4491
+     */
+    componentPath: string;
+    /**
+     * Container name, if known. This could be used to eagerly prefetch container code prior to having a snapshot.
+     */
+    containerCodeHint?: string;
```

Comment:
it's not clear what containerCodeHint is is all cases, maybe codeDetailsHint, with the format tdb by the driver

Reply:
Yeah, codeDetailsHint is exactly what this should be. It's a little weird that the driver is determining what the hint is when the codeloader is the one that will have to interpret it, but this can be thought more about when taking the codeDetailsHint implementation outside of just the odsp driver and would be out of scope for this PR. (I imagine the codeloader perhaps being the mechanism for providing the code hint etc.)